open-banking-accelerator/internal-apis/internal-webapps/com.wso2.openbanking.accelerator.event.notifications.endpoint/src/main/java/com/wso2/openbanking/accelerator/event/notifications/endpoint/api/EventPollingEndpoint.java

```diff
@@ -83,37 +83,43 @@ public Response pollEvents(@Context HttpServletRequest request, @Context HttpSer
             eventPollingData = parameterMap.get(EventNotificationEndPointConstants.REQUEST).
                     toString().replaceAll("\\\\r|\\\\n|\\r|\\n|\\[|]| ", StringUtils.EMPTY);
 
-            byte[] decodedBytes = Base64.getDecoder().decode(eventPollingData);
-            String decodedString = new String(decodedBytes, StandardCharsets.UTF_8);
-            try {
-                eventPollingRequest = (JSONObject) new JSONParser(JSONParser.MODE_PERMISSIVE).parse(decodedString);
-
-                //check if the client id is present in the header
-                String clientId = request.getHeader(EventNotificationConstants.X_WSO2_CLIENT_ID);
-                if (!StringUtils.isBlank(clientId)) {
-                    eventPollingRequest.put(EventNotificationConstants.X_WSO2_CLIENT_ID, request.
-                            getHeader(EventNotificationConstants.X_WSO2_CLIENT_ID));
-                } else {
+            if (!eventPollingData.isEmpty()) {
```

Comment:
Shall we use StringUtils.isBlank() check to avoid any null pointer exceptions?

Reply:
Fixed in https://github.com/wso2/financial-open-banking/pull/33/commits/cde0ef9d3216ab6b595d98a53eb3cd4006300d35